src/helpers.ts

```diff
@@ -37,3 +38,41 @@ export function sleep(ms: number, signal?: AbortSignal): Promise<void> {
     }
   });
 }
+
+/**
+ * Helper function to parse relative or ISO time strings
+ * @param timeStr - Examples: 'now', 'now-1d', 'now-3h', '2024-11-01T15:16:53.891Z'
+ * @returns Date object or null if parsing failed
+ */
+export function parseTime(timeStr: string): Date | null {
+  if (timeStr === 'now') {
+    return new Date();
+  }
+
+  if (timeStr.startsWith('now-')) {
+    const relativeMatch = timeStr.match(/now-(\d+)(s|m|h|d|w|mo|y)/i);
```

Comment:
![Screenshot 2024-11-05 at 14 01 51](https://github.com/user-attachments/assets/2be6c157-0ff5-4bff-b272-601b56b5e2a8)
Regex `mo` needs greedy, or be before `m`

Reply:
Nice find!